### ISSUE TITLE
[Suggestion] The use of 2-5-10 km eggs of choice.

### ISSUE THREAD

**Issue by Fenixxia** [NecronomiconCoding/NecroBot#521]
I wish you could choose to use eggs 2 and 5 km. Еhus we maximize the chance of getting eggs 10 km as this will increase exping per hour. The eggs for 10 km, you can save and use at higher levels.Thank you.

**Comment by KAustin1** [comment]
if you obtain an 10km egg with a level 8 trainer and hatch as level 20, it is a level 8 pokemon

**Comment by Fenixxia** [comment]
Do You have proof?

**Comment by KAustin1** [comment]
i am a level 22... I know for a fact.

**Comment by cloudsstars** [comment]
It appears that an egg's CP is based on your level when you found the egg, not when it hatches. We currently only have anecdotal evidence for this, but there is quite a lot of it, which I'll list at the end.

However, eggs will never hatch with a CP above trainer level 20. This was determined from reading the game's internal data files.

**Comment by VanStorm** [comment]
Might be a little off topic but does the current version automatically hatch eggs?

**Comment by cloudsstars** [comment]
@VanStorm been wondering that my self and how to set up if not. i cant tell if my eggs hatch or get deleted or if i just miss it on the recent list because it was low lvl and bot deleted it

**Comment by VanStorm** [comment]
@cloudsstars i know that it hatches those that you activate manually but i think that most of them get deleted since they aren't good enough but i've been asking myself if it automatically selects new eggs to hatch them once it's done hatching the one i incubated.

**Comment by cloudsstars** [comment]
@VanStorm thanks for the info i will hatch my 10k eggs now . i dont know if it auto adds i dont think it does as i have to add my eggs manually but that would be an awesome  to add if its not already

**Comment by Fenixxia** [comment]
Even if LP pokemon from eggs tied to the level of getting eggs, you need to add the function of eggs in any case. Need to free a slot for new eggs. There will be more of exping, high-level eggs too interesting.

**Comment by Bogenschuetze** [comment]
I implemented the UseAllIncubators Task which actually puts the eggs in the incubators automatically, but i dont know how to do a pull request

**Comment by lorenzowalz** [comment]
@Bogenschuetze https://yangsu.github.io/pull-request-tutorial/

**Comment by VanStorm** [comment]
@Bogenschuetze it would be really nice if you'd do a pull request with this feature.

**Comment by cawk** [comment]
i am currently working on this incase he doesnt make a pull request 
would people prefer to be able to choose what eggs they would like to incubate or wouldnt they mind?

**Comment by zzsnoozezzEB** [comment]
IncubatorUnlimited should prioritize 2km eggs while the BasicIncubators should do 10km and then 5km eggs

**Comment by cawk** [comment]
alright i can implement that i may add a feature so they can choose what eggs go in the unlimited one and if the basic is available what eggs go in that 
is that okay?

**Comment by Bogenschuetze** [comment]
Ill drop some code here, still have no clue how to do a pull request xD

**Comment by cawk** [comment]
Oh you have completed your code mind if i use this and make the changes i stated above about users being able to choose what eggs go in what types go in what incubators?

**Comment by Bogenschuetze** [comment]
Of course you can use it. I guess improvement is needed, i just implemented it for testing purposes :)

**Comment by cawk** [comment]
yeah theres a couple of things i will change but its very nice code well done

**Comment by cawk** [comment]
Since im going to be unavailable for around 48 hours i will push Bogenschuetze's code as it is for now and will submit my changes when i am back since my code is basically finished just got final few things to do

**Comment by lorenzowalz** [comment]
@cawk  @Bogenschuetze 

Keep in mind that the push request has a comment asking for revisions.
